library/Translate/src/Juvix/Contextify/Environment.hs

```diff
@@ -219,6 +291,16 @@ lookupPrecedence name ctx = do
 -- searchAndClosure function dispatch table
 ------------------------------------------------------------
 
+primitive :: HasClosure m => Sexp.T -> (Sexp.T -> m Sexp.T) -> m Sexp.T
+primitive (Sexp.List [name]) cont
+  | Just Sexp.A {atomName} <- Sexp.atomFromT name = do
+    -- this is bad, however it's FINE, as we only have the primitive
+    -- which no operation should disrupt
+    local @"closure" (Closure.insertGeneric (NameSymbol.hd atomName)) $ do
+      prim <- cont name
+      pure $ Sexp.list [prim]
```

Comment:
sorry could you clarify what could (or couldn't) go wrong here?

Reply:
So we add the hd of the name so in 
```lisp
(:primitive Michelson.Foo)
```
we add Michelson to the closure, which is wrong.

However, since we just have the primitive at this point, we can't have any other symbol named Michelson in this scope, so it's not an actual issue.